### PR TITLE
Adding sample finetuning scripts to guide the users towards fine-tuning

### PR DESCRIPTION
**Pull request by agautam478**
Added two sample fine-tuning scripts to enable fine-tuning.
Users can run them as-is if they have:

  - dependencies installed with uv sync
  - Hugging Face access/token for the published weights
  - CSVs with enough rows for seq_len + forecast_horizon in forecasting, or window_length in AD
  - numeric feature columns; AD inference should drop timestamp/label metadata before calling the SDK

### CODE REVIEW

File: ad_diffusion/examples/finetune_example.py [hunk 1]
```diff
@@ -0,0 +1,443 @@
...
+    parser.add_argument("--num-workers", type=int, default=0)
+    parser.add_argument("--seed", type=int, default=42)
+    parser.add_argument("--output-dir", default="artifacts/finetune")
+
+    parser.add_argument("--window-length", type=int, default=None)
+    parser.add_argument("--window-stride", type=int, default=1)
+    parser.add_argument("--split", type=int, default=None, help="Number of alternating mask segments per window.")
+    parser.add_argument("--mask-ratio", type=float, default=0.7, help="Random masking ratio used during training.")
+    parser.add_argument("--scale-factor", type=float, default=None)
+    parser.add_argument("--val-ratio", type=float, default=0.1)
```
Comment (neptunel) [inline ad_diffusion/examples/finetune_example.py]: I ran both fine-tuning scripts for one epoch. The forecasting fine-tune completed successfully and saved checkpoints under artifacts/. The AD diffusion run failed because the default --val-ratio 0.1 on the 500-row test dataset creates only 50 validation rows, which is less than the default --window-length 100. Re-running with --val-ratio 0.3 (or --window-length <=50) resolves the issue. It would be helpful to document this in the README or change the deafult/ sample dataset size

**Review by neptunel — APPROVED** [review]
